crates/circuit/src/operations.rs

```diff
@@ -741,13 +1072,15 @@ impl StandardGate {
     pub fn create_py_op(
         &self,
         py: Python,
-        params: Option<&[Param]>,
+        params: Option<SmallVec<[Param; 3]>>,
         label: Option<&str>,
     ) -> PyResult<Py<PyAny>> {
         let gate_class = get_std_gate_class(py, *self)?;
-        let args = match params.unwrap_or(&[]) {
-            &[] => PyTuple::empty(py),
-            params => PyTuple::new(py, params.iter().map(|x| x.into_pyobject(py).unwrap()))?,
+        let args = match params {
+            None => PyTuple::empty(py),
+            Some(params) => {
+                PyTuple::new(py, params.into_iter().map(|x| x.into_pyobject(py).unwrap()))?
+            }
```

Comment:
It shouldn't be on the hot path, but this feels like it's become less efficient - we don't actually use the ownership of the `params` object, so I don't think there's a benefit to the two places that naturally produced a `SmallVec`, and there's a new cost inserted to the two places that natively had a `&[Param]`.

Reply:
Ah alright. I was assuming that `IntoPyObject` on `&Param` would clone (it does for `ParameterExpression`s), but I see now that `IntoPyObject` on `Param` itself actually just calls the ref impl.